Source/ACE/Entity/Landblock.cs

```diff
@@ -543,19 +455,36 @@ public void UseTime()
                         }
                     });
 
-                    if (mo.Location.LandblockId == id)
-                    {
-                        // update if it's still here
-                        Broadcast(BroadcastEventArgs.CreateAction(BroadcastAction.AddOrUpdate, mo), true, Quadrant.All);
-                    }
-                    else
+                    if (mo.Location.LandblockId != id)
                     {
                         // remove and readd if it's not
                         RemoveWorldObject(mo.Guid, false);
                         LandblockManager.AddObject(mo);
                     }
                 });
 
+                // Tick all objects
+                double tickTime = WorldManager.PortalYearTicks;
+                // per-creature update on landblock.
+                Parallel.ForEach(allworldobj, wo =>
+                {
+                    // Process the creatures
+                    // We lock here -- it IS required becuase this is run in a Parallel.ForEach loop
+                    //   AND because of cross-landblock dependencies
+                    //   Any use of OTHER objects within the tick function (gotten through landblock by guid) requires locking
+                    //   updates of THIS object within Tick are safe
+                    var tickable = wo as ITickable;
+                    if (tickable != null)
+                    {
+                        lock (tickable)
+                        {
+                            // FIXME(ddevec): LBBroadcaster class was a fast hacky fix to the problem -- there is probably a more elegant solution..
```

Comment:
I'm not sure if I understand how locking inside of a Parallel.ForEach statement is useless.  The objects within the loop may be interacting with each other, e.g. a monster may attack a player and adjust her health, and the player may be regenerating that health.   Unless Parallel.ForEach does some black magic, that would result in a data-race, requiring locking.

I didn't think of the danger of deadlock when I wrote this code.  That is a very important point and I don't have a method of mitigating it.  Mitigating lock-based deadlocks is traditionally done by enforcing lock ordering, which would be difficult to do with this design...